# - Dermstore - 
# Analyse exploratoire des données 

### Ce dataset présente des produits cosmétiques issus du site Dermstore et comprend des détails sur chaque produit tels que leurs caractéristiques mais aussi la notation, le stock, le chemin d'accès à l'image produit et d'autres données liées au scrapping.

### L'analyse de ce dataset s'inscrit dans le cadre du projet d'entraînement d'un modèle permettant de proposer des produits adaptés à des problématiques de peau après soumission d'une photo selfie par l'utilisateur. Il servira de base pour l'affichage des produits recommandés.

## Import des données

In [3]:
import pandas as pd
import json

In [4]:
df = pd.read_json("dermstore_data.json")

In [6]:
df.to_csv("dermstore_data.csv", index=False, encoding="utf-8")

## Analyse générale du dataset

In [7]:
print("Nombre de lignes : ", df.shape[0])
print("Nombre de colonnes : ", df.shape[1])

Nombre de lignes :  126
Nombre de colonnes :  24


In [8]:
# aperçu des 5 premières lignes
df.head(5)

,url,title,brand,sku,product_id,price,currency,availability,condition,rating_value,...,raw_ingredients,how_to_use,raw_how_to_use,skin_type_and_concerns,raw_skin_type_and_concerns,range,volume,images,uniq_id,scraped_at
0,https://www.dermstore.com/p/exclusive-neostrat...,Exclusive NEOSTRATA Anti-Aging Firming Duo,Neostrata,13441438,13441438,188.0,USD,OutOfStock,NewCondition,,...,"<div aria-labelledby=""Ingredients"" class=""cont...",Gently smooth over neck and décolletage twice ...,"<div aria-labelledby=""How-to-Use"" class=""conte...",,,,80ml,https://static.thcdn.com/productimg/original/1...,6635eafe-d8bd-53c7-b232-d359b208b4a1,2024-02-11 11:15:28
1,https://www.dermstore.com/p/pmd-pmd-clean-1-pi...,PMD Clean (1 piece),PMD,12905594,12905594,79.2,USD,InStock,NewCondition,4.38,...,"<div aria-labelledby=""Ingredients"" class=""cont...",Step 1: Rinse and Apply Cleanser.\n Rinse fac...,"<div aria-labelledby=""How-to-Use"" class=""conte...",Acne: BlackheadsAging Skin: Loss of FirmnessAp...,"<div aria-labelledby=""Skin-Type-&amp;-Concerns...",,,https://static.thcdn.com/productimg/original/1...,7c436149-00ab-56d5-8274-cf24a991dbaf,2024-02-11 11:15:29
2,https://www.dermstore.com/p/neom-focus-the-min...,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,NEOM,12540840,12540840,24.0,USD,InStock,NewCondition,,...,"<div aria-labelledby=""Ingredients"" class=""cont...",To experience the full Feel Good Vibes Essenti...,"<div aria-labelledby=""How-to-Use"" class=""conte...",,,,10ml,https://static.thcdn.com/productimg/original/1...,6c9186b2-c19d-55e5-a45b-c6be55bc74d6,2024-02-11 11:15:33
3,https://www.dermstore.com/p/briogeo-curl-chari...,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo,11271963,11271963,28.0,USD,InStock,NewCondition,4.33,...,"<div aria-labelledby=""Ingredients"" class=""cont...",Apply to wet hair and massage from root to tip...,"<div aria-labelledby=""How-to-Use"" class=""conte...","Application Area: Hair, ScalpIdeal for these C...","<div aria-labelledby=""Skin-Type-&amp;-Concerns...",Curl Charisma,236ml,https://static.thcdn.com/productimg/original/1...,2b54ea9f-b05a-58be-8ef6-37d115ac19a9,2024-02-11 11:15:34
4,https://www.dermstore.com/p/dermalogica-christ...,Dermalogica Retinol Renewal Gift Set ($186 Value),Dermalogica,15508780,15508780,129.0,USD,InStock,NewCondition,,...,"<div aria-labelledby=""Ingredients"" class=""cont...",Super Rich Repair: Apply to face and neck with...,"<div aria-labelledby=""How-to-Use"" class=""conte...",,,,,https://static.thcdn.com/productimg/original/1...,57eafd78-cb46-57fb-9ff1-564f8f47f41d,2024-02-11 11:15:34


In [9]:
# types de données
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 126 entries, 0 to 125
Data columns (total 24 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   url                         126 non-null    str           
 1   title                       126 non-null    str           
 2   brand                       126 non-null    str           
 3   sku                         126 non-null    int64         
 4   product_id                  126 non-null    int64         
 5   price                       126 non-null    float64       
 6   currency                    126 non-null    str           
 7   availability                126 non-null    str           
 8   condition                   126 non-null    str           
 9   rating_value                126 non-null    object        
 10  review_count                126 non-null    object        
 11  category                    126 non-null    str           
 12  descr

In [10]:
df.describe(include="all")

,url,title,brand,sku,product_id,price,currency,availability,condition,rating_value,...,raw_ingredients,how_to_use,raw_how_to_use,skin_type_and_concerns,raw_skin_type_and_concerns,range,volume,images,uniq_id,scraped_at
count,126,126,126,1.260000e+02,1.260000e+02,126.000000,126,126,126,126,...,126,126,126,126,126,126,126,126,126,126
unique,126,126,86,NaN,NaN,NaN,1,2,1,32,...,121,124,124,54,54,45,43,126,126,NaN
top,https://www.dermstore.com/p/exclusive-neostrat...,Exclusive NEOSTRATA Anti-Aging Firming Duo,Miriam Quevedo,NaN,NaN,NaN,USD,InStock,NewCondition,,...,,,,,,,,https://static.thcdn.com/productimg/original/1...,6635eafe-d8bd-53c7-b232-d359b208b4a1,NaN
freq,1,1,4,NaN,NaN,NaN,126,107,126,66,...,6,3,3,73,73,82,62,1,1,NaN
mean,NaN,NaN,NaN,1.318559e+07,1.318559e+07,78.088333,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2024-02-11 11:17:24.230158
min,NaN,NaN,NaN,1.030272e+07,1.030272e+07,11.000000,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2024-02-11 11:15:28
25%,NaN,NaN,NaN,1.169451e+07,1.169451e+07,35.000000,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2024-02-11 11:16:21
50%,NaN,NaN,NaN,1.291211e+07,1.291211e+07,52.500000,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2024-02-11 11:17:18
75%,NaN,NaN,NaN,1.445992e+07,1.445992e+07,97.250000,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2024-02-11 11:18:28.750000
max,NaN,NaN,NaN,1.564870e+07,1.564870e+07,495.000000,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2024-02-11 11:20:15


Le dataset contient 126 produits issus de plus de 80 marques distinctes.
Les descriptions sont complètes dans la majorité des cas, tandis que les ingrédients présentent quelques valeurs manquantes.
Certaines marques dominent le dataset, ce qui peut introduire un biais dans les recommandations.
Les produits couvrent une large gamme de prix.

In [11]:
df.duplicated().sum() # on vérifie les doublons

np.int64(0)

In [12]:
df["skin_type_and_concerns"].value_counts()

skin_type_and_concerns
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            73
Acne: BlackheadsAging Skin: Loss of FirmnessApplication Area: FaceIdeal for these Concerns: Acne, Large Pores, OilinessProfessional Tools and Devices: Cleansing DevicesSkin Care: Professional Tools and DevicesSkin Type: Acne-Prone Skin, Combination Skin, Dry Skin, Mature Skin, Normal Skin, Oily Skin, Sensitive Skin                                                                                                      

In [13]:
df["category"] # plusieurs types de données se trouvent dans "category" -> il faudra laisser supprimer les données incohérentes

0      Brands / NEOSTRATA / Exclusive NEOSTRATA Anti-...
1                     Brands / PMD / PMD Clean (1 piece)
2      Brands / NEOM / NEOM Focus the Mind Essential ...
3      Brands / Briogeo / Briogeo Curl Charisma™ Rice...
4      Brands / Dermalogica / Dermalogica Retinol Ren...
                             ...                        
121    Brands / RMS Beauty / RMS Beauty Kakadu Beauty...
122    Brands / Tarte Cosmetics / Tarte Foundcealer F...
123    Brands / Erborian / Erborian Skin Therapy Mult...
124    Brands / This Works / this works Perfect Hands...
125    Brands / KYPRIS / KYPRIS Clearing Serum Balanc...
Name: category, Length: 126, dtype: str

In [14]:
df.columns

Index(['url', 'title', 'brand', 'sku', 'product_id', 'price', 'currency',
       'availability', 'condition', 'rating_value', 'review_count', 'category',
       'description', 'ingredients', 'raw_ingredients', 'how_to_use',
       'raw_how_to_use', 'skin_type_and_concerns',
       'raw_skin_type_and_concerns', 'range', 'volume', 'images', 'uniq_id',
       'scraped_at'],
      dtype='str')

## Nettoyage des données

### Suppression des colonnes inutiles

In [15]:
df.drop(["url","uniq_id", "scraped_at", "availability", "sku", "product_id", "price", "currency", "raw_ingredients", "raw_how_to_use", "raw_skin_type_and_concerns", "condition", "how_to_use", "range", "volume", "rating_value", "review_count"], axis=1, inplace=True)


In [16]:
df.isnull().sum()

title                     0
brand                     0
category                  0
description               0
ingredients               0
skin_type_and_concerns    0
images                    0
dtype: int64

In [17]:
(df == "").sum()

title                      0
brand                      0
category                   0
description                0
ingredients                6
skin_type_and_concerns    73
images                     0
dtype: int64

Les valeurs cachées sont au nombre de 6 pour "ingredients" et 73 pour "skin_type_and_concerns".

In [18]:
df["category"] = df["category"].str.split("/").str[-1].str.strip()

La colonne "category" présente des informations incohérentes son intitulé : les 2 premiers éléments, "Brands" et la marque séparés par des / de manière à ne laisser que le nom du produit.

In [19]:
df

,title,brand,category,description,ingredients,skin_type_and_concerns,images
0,Exclusive NEOSTRATA Anti-Aging Firming Duo,Neostrata,Exclusive NEOSTRATA Anti-Aging Firming Duo,This anti aging neck cream is formulated with...,"Aqua (Water), Acetyl Glucosamine, Cyclopentasi...",,https://static.thcdn.com/productimg/original/1...
1,PMD Clean (1 piece),PMD,PMD Clean (1 piece),Give yourself a mini facial with help from PM...,Silicone AA Battery.\nFor the latest informati...,Acne: BlackheadsAging Skin: Loss of FirmnessAp...,https://static.thcdn.com/productimg/original/1...
2,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,NEOM,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,Bringing you back to the present moment in a ...,"Caprylic/Capric Triglyceride, Eucalyptus Globu...",,https://static.thcdn.com/productimg/original/1...
3,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo's Curl Charisma™ Rice Amino + Avocado...,"Water/Aqua/Eau, Lauramidopropyl Betaine*, Sodi...","Application Area: Hair, ScalpIdeal for these C...",https://static.thcdn.com/productimg/original/1...
4,Dermalogica Retinol Renewal Gift Set ($186 Value),Dermalogica,Dermalogica Retinol Renewal Gift Set ($186 Value),Renew skin with a transformative duo that tra...,"Super Rich Repair: Water/Aqua/Eau, Isohexadeca...",,https://static.thcdn.com/productimg/original/1...
...,...,...,...,...,...,...,...
121,RMS Beauty Kakadu Beauty Oil 30ml,RMS Beauty,RMS Beauty Kakadu Beauty Oil 30ml,"Longwearing, crease-proof cream eyeshadows in...","*Simmondsia Chinensis (Jojoba) Seed Oil, *Term...",,https://static.thcdn.com/productimg/original/1...
122,Tarte Foundcealer Foundation Brush (1 piece),Tarte,Tarte Foundcealer Foundation Brush (1 piece),"Press and buff on comfortable, flawless mediu...","N/A\nFor the latest information, it is recomme...",Application Area: FaceCosmetic Tools: Makeup B...,https://static.thcdn.com/productimg/original/1...
123,Erborian Skin Therapy Multi-Perfecting Night O...,Erborian,Erborian Skin Therapy Multi-Perfecting Night O...,Skin Therapy is a new generation of face oil-...,Aqua/Water - Glycerin - Caprylic/Capric Trigly...,,https://static.thcdn.com/productimg/original/1...
124,this works Perfect Hands Intense Moisture (75ml),this works,this works Perfect Hands Intense Moisture (75ml),Keep your hands looking young with the this w...,"Aqua, Glyceryl Stearate SE, Glycerin, Cetearyl...",,https://static.thcdn.com/productimg/original/1...


### Suppression des retours à la ligne et des espaces

In [20]:
import re

def clean_text(text):
    if pd.isna(text):
        return ""
    text = str(text)
    text = re.sub(r"\s+", " ", text)  # espaces multiples
    return text.strip()

for col in df.columns:
    df[col] = df[col].apply(clean_text)

In [21]:
df

,title,brand,category,description,ingredients,skin_type_and_concerns,images
0,Exclusive NEOSTRATA Anti-Aging Firming Duo,Neostrata,Exclusive NEOSTRATA Anti-Aging Firming Duo,This anti aging neck cream is formulated with ...,"Aqua (Water), Acetyl Glucosamine, Cyclopentasi...",,https://static.thcdn.com/productimg/original/1...
1,PMD Clean (1 piece),PMD,PMD Clean (1 piece),Give yourself a mini facial with help from PMD...,Silicone AA Battery. For the latest informatio...,Acne: BlackheadsAging Skin: Loss of FirmnessAp...,https://static.thcdn.com/productimg/original/1...
2,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,NEOM,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,Bringing you back to the present moment in a s...,"Caprylic/Capric Triglyceride, Eucalyptus Globu...",,https://static.thcdn.com/productimg/original/1...
3,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo's Curl Charisma™ Rice Amino + Avocado ...,"Water/Aqua/Eau, Lauramidopropyl Betaine*, Sodi...","Application Area: Hair, ScalpIdeal for these C...",https://static.thcdn.com/productimg/original/1...
4,Dermalogica Retinol Renewal Gift Set ($186 Value),Dermalogica,Dermalogica Retinol Renewal Gift Set ($186 Value),Renew skin with a transformative duo that tran...,"Super Rich Repair: Water/Aqua/Eau, Isohexadeca...",,https://static.thcdn.com/productimg/original/1...
...,...,...,...,...,...,...,...
121,RMS Beauty Kakadu Beauty Oil 30ml,RMS Beauty,RMS Beauty Kakadu Beauty Oil 30ml,"Longwearing, crease-proof cream eyeshadows in ...","*Simmondsia Chinensis (Jojoba) Seed Oil, *Term...",,https://static.thcdn.com/productimg/original/1...
122,Tarte Foundcealer Foundation Brush (1 piece),Tarte,Tarte Foundcealer Foundation Brush (1 piece),"Press and buff on comfortable, flawless medium...","N/A For the latest information, it is recommen...",Application Area: FaceCosmetic Tools: Makeup B...,https://static.thcdn.com/productimg/original/1...
123,Erborian Skin Therapy Multi-Perfecting Night O...,Erborian,Erborian Skin Therapy Multi-Perfecting Night O...,Skin Therapy is a new generation of face oil-s...,Aqua/Water - Glycerin - Caprylic/Capric Trigly...,,https://static.thcdn.com/productimg/original/1...
124,this works Perfect Hands Intense Moisture (75ml),this works,this works Perfect Hands Intense Moisture (75ml),Keep your hands looking young with the this wo...,"Aqua, Glyceryl Stearate SE, Glycerin, Cetearyl...",,https://static.thcdn.com/productimg/original/1...


### Supprimer le bruit marketing inutile

In [22]:
def remove_noise(text):
    noise_patterns = [
        r"For the latest information.*",
        r"<em>.*?</em>"
    ]
    for pattern in noise_patterns:
        text = re.sub(pattern, "", text, flags=re.IGNORECASE)
    return text

In [23]:
df

,title,brand,category,description,ingredients,skin_type_and_concerns,images
0,Exclusive NEOSTRATA Anti-Aging Firming Duo,Neostrata,Exclusive NEOSTRATA Anti-Aging Firming Duo,This anti aging neck cream is formulated with ...,"Aqua (Water), Acetyl Glucosamine, Cyclopentasi...",,https://static.thcdn.com/productimg/original/1...
1,PMD Clean (1 piece),PMD,PMD Clean (1 piece),Give yourself a mini facial with help from PMD...,Silicone AA Battery. For the latest informatio...,Acne: BlackheadsAging Skin: Loss of FirmnessAp...,https://static.thcdn.com/productimg/original/1...
2,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,NEOM,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,Bringing you back to the present moment in a s...,"Caprylic/Capric Triglyceride, Eucalyptus Globu...",,https://static.thcdn.com/productimg/original/1...
3,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo's Curl Charisma™ Rice Amino + Avocado ...,"Water/Aqua/Eau, Lauramidopropyl Betaine*, Sodi...","Application Area: Hair, ScalpIdeal for these C...",https://static.thcdn.com/productimg/original/1...
4,Dermalogica Retinol Renewal Gift Set ($186 Value),Dermalogica,Dermalogica Retinol Renewal Gift Set ($186 Value),Renew skin with a transformative duo that tran...,"Super Rich Repair: Water/Aqua/Eau, Isohexadeca...",,https://static.thcdn.com/productimg/original/1...
...,...,...,...,...,...,...,...
121,RMS Beauty Kakadu Beauty Oil 30ml,RMS Beauty,RMS Beauty Kakadu Beauty Oil 30ml,"Longwearing, crease-proof cream eyeshadows in ...","*Simmondsia Chinensis (Jojoba) Seed Oil, *Term...",,https://static.thcdn.com/productimg/original/1...
122,Tarte Foundcealer Foundation Brush (1 piece),Tarte,Tarte Foundcealer Foundation Brush (1 piece),"Press and buff on comfortable, flawless medium...","N/A For the latest information, it is recommen...",Application Area: FaceCosmetic Tools: Makeup B...,https://static.thcdn.com/productimg/original/1...
123,Erborian Skin Therapy Multi-Perfecting Night O...,Erborian,Erborian Skin Therapy Multi-Perfecting Night O...,Skin Therapy is a new generation of face oil-s...,Aqua/Water - Glycerin - Caprylic/Capric Trigly...,,https://static.thcdn.com/productimg/original/1...
124,this works Perfect Hands Intense Moisture (75ml),this works,this works Perfect Hands Intense Moisture (75ml),Keep your hands looking young with the this wo...,"Aqua, Glyceryl Stearate SE, Glycerin, Cetearyl...",,https://static.thcdn.com/productimg/original/1...


### Nettoyage des ingrédients

In [24]:
def clean_ingredients(text):
    if not text:
        return ""
    # enlever doublons simples
    ingredients = list(set([i.strip() for i in text.split(",")]))
    return ", ".join(ingredients)

df["ingredients"] = df["ingredients"].apply(clean_ingredients)

In [25]:
df

,title,brand,category,description,ingredients,skin_type_and_concerns,images
0,Exclusive NEOSTRATA Anti-Aging Firming Duo,Neostrata,Exclusive NEOSTRATA Anti-Aging Firming Duo,This anti aging neck cream is formulated with ...,it is recommended to review the ingredient lis...,,https://static.thcdn.com/productimg/original/1...
1,PMD Clean (1 piece),PMD,PMD Clean (1 piece),Give yourself a mini facial with help from PMD...,Silicone AA Battery. For the latest informatio...,Acne: BlackheadsAging Skin: Loss of FirmnessAp...,https://static.thcdn.com/productimg/original/1...
2,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,NEOM,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,Bringing you back to the present moment in a s...,"Pinus Sylvestris Leaf Oil, Limonene, Juniperus...",,https://static.thcdn.com/productimg/original/1...
3,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo's Curl Charisma™ Rice Amino + Avocado ...,"Cocamidopropyl Hydroxysultaine*, it is recomme...","Application Area: Hair, ScalpIdeal for these C...",https://static.thcdn.com/productimg/original/1...
4,Dermalogica Retinol Renewal Gift Set ($186 Value),Dermalogica,Dermalogica Retinol Renewal Gift Set ($186 Value),Renew skin with a transformative duo that tran...,"Yeast Extract, Citronellol, Sodium Benzoate, B...",,https://static.thcdn.com/productimg/original/1...
...,...,...,...,...,...,...,...
121,RMS Beauty Kakadu Beauty Oil 30ml,RMS Beauty,RMS Beauty Kakadu Beauty Oil 30ml,"Longwearing, crease-proof cream eyeshadows in ...","Artemisia Vulgaris (Mugwort) Oil, it is recomm...",,https://static.thcdn.com/productimg/original/1...
122,Tarte Foundcealer Foundation Brush (1 piece),Tarte,Tarte Foundcealer Foundation Brush (1 piece),"Press and buff on comfortable, flawless medium...",it is recommended to review the ingredient lis...,Application Area: FaceCosmetic Tools: Makeup B...,https://static.thcdn.com/productimg/original/1...
123,Erborian Skin Therapy Multi-Perfecting Night O...,Erborian,Erborian Skin Therapy Multi-Perfecting Night O...,Skin Therapy is a new generation of face oil-s...,it is recommended to review the ingredient lis...,,https://static.thcdn.com/productimg/original/1...
124,this works Perfect Hands Intense Moisture (75ml),this works,this works Perfect Hands Intense Moisture (75ml),Keep your hands looking young with the this wo...,"Benzyl Alcohol, Limonene, Parfum, Glycerin, Ca...",,https://static.thcdn.com/productimg/original/1...


### Gestion des valeurs manquantes

In [26]:
df.fillna("", inplace=True)

,title,brand,category,description,ingredients,skin_type_and_concerns,images
0,Exclusive NEOSTRATA Anti-Aging Firming Duo,Neostrata,Exclusive NEOSTRATA Anti-Aging Firming Duo,This anti aging neck cream is formulated with ...,it is recommended to review the ingredient lis...,,https://static.thcdn.com/productimg/original/1...
1,PMD Clean (1 piece),PMD,PMD Clean (1 piece),Give yourself a mini facial with help from PMD...,Silicone AA Battery. For the latest informatio...,Acne: BlackheadsAging Skin: Loss of FirmnessAp...,https://static.thcdn.com/productimg/original/1...
2,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,NEOM,NEOM Focus the Mind Essential Oil Blend 0.3 fl...,Bringing you back to the present moment in a s...,"Pinus Sylvestris Leaf Oil, Limonene, Juniperus...",,https://static.thcdn.com/productimg/original/1...
3,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo,Briogeo Curl Charisma™ Rice Amino + Avocado Hy...,Briogeo's Curl Charisma™ Rice Amino + Avocado ...,"Cocamidopropyl Hydroxysultaine*, it is recomme...","Application Area: Hair, ScalpIdeal for these C...",https://static.thcdn.com/productimg/original/1...
4,Dermalogica Retinol Renewal Gift Set ($186 Value),Dermalogica,Dermalogica Retinol Renewal Gift Set ($186 Value),Renew skin with a transformative duo that tran...,"Yeast Extract, Citronellol, Sodium Benzoate, B...",,https://static.thcdn.com/productimg/original/1...
...,...,...,...,...,...,...,...
121,RMS Beauty Kakadu Beauty Oil 30ml,RMS Beauty,RMS Beauty Kakadu Beauty Oil 30ml,"Longwearing, crease-proof cream eyeshadows in ...","Artemisia Vulgaris (Mugwort) Oil, it is recomm...",,https://static.thcdn.com/productimg/original/1...
122,Tarte Foundcealer Foundation Brush (1 piece),Tarte,Tarte Foundcealer Foundation Brush (1 piece),"Press and buff on comfortable, flawless medium...",it is recommended to review the ingredient lis...,Application Area: FaceCosmetic Tools: Makeup B...,https://static.thcdn.com/productimg/original/1...
123,Erborian Skin Therapy Multi-Perfecting Night O...,Erborian,Erborian Skin Therapy Multi-Perfecting Night O...,Skin Therapy is a new generation of face oil-s...,it is recommended to review the ingredient lis...,,https://static.thcdn.com/productimg/original/1...
124,this works Perfect Hands Intense Moisture (75ml),this works,this works Perfect Hands Intense Moisture (75ml),Keep your hands looking young with the this wo...,"Benzyl Alcohol, Limonene, Parfum, Glycerin, Ca...",,https://static.thcdn.com/productimg/original/1...


### Ce dataset est globalement bien adapté à une utilisation en RAG, à condition d'avoir effectué au préalable un prétraitement léger maus utile.

In [27]:
df.to_csv("dermstore_skincare_cleaned.csv", index=False)